### PR TITLE
Modified emit_isk12.

### PR DESCRIPTION
**Pull request by sindrom91**
I redesigned `emit_isk12` to better fit its purpose (in my opinion). Here are some reasons I found for redesign:
 - Error code was returned as -1 (signed value), even though the return value is `uint32_t`
 - Data-processing k12 cannot handle negative numbers
 - Handling 64-bit constants

What I did:
 - Handling negative constants (`add x1, x1, -8` (invalid) --> `sub x1, x1, 8`)
 - `A64I_BINOPk` is appended to the returned constant (just like in ARM32)
 - Encoded constant is returned by reference
 - Return value is just an indicator of success

Please let me know if you have any comments.

### CODE REVIEW

File: src/lj_asm_arm64.h [hunk 1]
```diff
@@ -1899,8 +1908,10 @@ static void asm_tail_fixup(ASMState *as, TraceNo lnk)
     as->mctop = --p;
   } else {
     /* Patch stack adjustment. */
-    uint32_t k = emit_isk12(A64I_ADDx, spadj);
-    p[-2] = (A64I_ADDx^A64I_BINOPk^k) | A64F_D(RID_SP) | A64F_N(RID_SP);
+    uint32_t k;
+    emit_isk12(spadj, &k);
+    lua_assert(isk12);
```
Comment (cbaylis) [inline src/lj_asm_arm64.h]: isk12 isn't defined here.

File: src/lj_emit_arm64.h [hunk 2]
```diff
@@ -56,17 +56,18 @@ static void emit_dnma(ASMState *as, A64Ins ai, Reg rd, Reg rn, Reg rm, Reg ra)
...
-    }
-    if ((n & 0xfff) == 0 && n < (0xfff << 12))
-    {
-        return (((n >> 12) & 0xfff) << 10) | (1 << 22);
-    }
-    return -1;
+  uint64_t k = (n < 0) ? -n : n;
+  uint32_t m = (n < 0) ? 0x40000000 : 0;
+  if (k < 4096) {
+    *k12 = A64I_BINOPk|m|(k<<10);
```
Comment (cbaylis) [inline src/lj_emit_arm64.h]: I think this should use ^m instead of |m so that SUB becomes an ADD.

File: src/lj_asm_arm64.h (hunk 1 above)
Comment (sindrom91) [inline src/lj_asm_arm64.h]: Oops, leftover.

File: src/lj_emit_arm64.h (hunk 2 above)
Comment (sindrom91) [inline src/lj_emit_arm64.h]: I'm not actually modifying the instruction here, I'm just adding that bit to the equation. Whoever uses `emit_isk12` needs to ^ the constant with the desired instruction.

**Pull request merged**